src/inference/src/dev/core_impl.cpp

```diff
@@ -877,7 +880,10 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
         CacheContent cacheContent{cacheManager, parsed._core_config.get_enable_mmap()};
         cacheContent.blobId = ov::ModelCache::compute_hash(model, create_compile_config(plugin, parsed._config));
         std::unique_ptr<CacheGuardEntry> lock = cacheGuard.get_hash_lock(cacheContent.blobId);
-        cacheContent.model = model;
+        if (parsed._config.find(ov::cache_mode.name()) != parsed._config.end() &&
+            parsed._config.at(ov::cache_mode.name()).as<ov::CacheMode>() == ov::CacheMode::OPTIMIZE_SIZE) {
+            cacheContent.model = model;
+        }
```

Comment:
```suggestion
        if (auto mode_it = parsed._config.find(ov::cache_mode.name()); mode_it != parsed._config.end()) {
            if (mode_it->second.as<ov::CacheMode>() == ov::CacheMode::OPTIMIZE_SIZE) {
                cacheContent.model = model;
            }
        }
```
Optional parsed config is checked once.

Reply:
I updated it as you reviewed. Thank you!